### PR TITLE
Remove posibility to export to onnx

### PR DESCRIPTION
**Pull request by mvafin**
### Details:
 - *Avoid possibility to auto-export to onnx in convert_model*

### Tickets:
 - *103726*

### CODE REVIEW

File: tools/mo/openvino/tools/mo/moc_frontend/pytorch_frontend_utils.py [hunk 1]
```diff
@@ -1,37 +1,25 @@
...
 
 def get_pytorch_decoder(model, input_shape, example_inputs, input_info):
     try:
         from openvino.frontend.pytorch.decoder import TorchScriptPythonDecoder
     except Exception as e:
         log.error("PyTorch frontend loading failed")
         raise e
-    inputs = prepare_torch_inputs(example_inputs, input_shape, input_info, allow_none=True)
+    inputs = prepare_torch_inputs(
+        example_inputs, input_shape, input_info, allow_none=True)
```
Comment (mvafin) [inline tools/mo/openvino/tools/mo/moc_frontend/pytorch_frontend_utils.py:20]:
```suggestion
    inputs = prepare_torch_inputs(example_inputs, input_shape, input_info, allow_none=True)
```

File: tools/mo/openvino/tools/mo/moc_frontend/pytorch_frontend_utils.py [hunk 2]
```diff
@@ -74,7 +64,8 @@ def get_torch_dtype(dtype):
         if str_dtype is None:
             raise Error(f"Unexpected data type '{dtype}' for input")
         return str_dtype
-    raise Error(f"Unexpected data type for input. Supporteed torch.dtype, numpy.dtype, ov.Type and str. Got {type(dtype)}")
+    raise Error(
+        f"Unexpected data type for input. Supported torch.dtype, numpy.dtype, ov.Type and str. Got {type(dtype)}")
```
Comment (mvafin) [inline tools/mo/openvino/tools/mo/moc_frontend/pytorch_frontend_utils.py:68]:
```suggestion
    raise Error(f"Unexpected data type for input. Supported torch.dtype, numpy.dtype, ov.Type and str. Got {type(dtype)}")
```

File: tools/mo/openvino/tools/mo/moc_frontend/pytorch_frontend_utils.py [hunk 3]
```diff
@@ -94,7 +85,8 @@ def prepare_torch_inputs(example_inputs, input_shape, input_info=None, allow_non
         elif isinstance(inputs, dict):
             for name, tensor in inputs.items():
                 assert isinstance(name, str), "Expected dictionary where keys are input names of string type and" \
-                                              " values are tensors. Got key of type {}".format(type(name))
+                                              " values are tensors. Got key of type {}".format(
+                                                  type(name))
```
Comment (mvafin) [inline tools/mo/openvino/tools/mo/moc_frontend/pytorch_frontend_utils.py:89]:
```suggestion
                                              " values are tensors. Got key of type {}".format(type(name))
```

File: tools/mo/openvino/tools/mo/moc_frontend/pytorch_frontend_utils.py [hunk 4]
```diff
@@ -106,7 +98,8 @@ def prepare_torch_inputs(example_inputs, input_shape, input_info=None, allow_non
             shape = inp.shape
             if shape is None:
                 if not allow_none:
-                    raise Error("Please provide input_shape or example_input for all inputs converting PyTorch model.")
+                    raise Error(
+                        "Please provide input_shape or example_input for all inputs converting PyTorch model.")
```
Comment (mvafin) [inline tools/mo/openvino/tools/mo/moc_frontend/pytorch_frontend_utils.py:102]:
```suggestion
                    raise Error("Please provide input_shape or example_input for all inputs converting PyTorch model.")
```

File: tools/mo/openvino/tools/mo/moc_frontend/pytorch_frontend_utils.py [hunk 5]
```diff
@@ -116,69 +109,6 @@ def prepare_torch_inputs(example_inputs, input_shape, input_info=None, allow_non
             inputs = tuple(inputs)
     else:
         if not allow_none:
-            raise Error("Please provide input_shape or example_input for converting PyTorch model.")
+            raise Error(
+                "Please provide input_shape or example_input for converting PyTorch model.")
```
Comment (mvafin) [inline tools/mo/openvino/tools/mo/moc_frontend/pytorch_frontend_utils.py:113]:
```suggestion
            raise Error("Please provide input_shape or example_input for converting PyTorch model.")
```

File: tools/mo/openvino/tools/mo/convert_impl.py [hunk 6]
```diff
@@ -873,12 +873,6 @@ def _convert(cli_parser: argparse.ArgumentParser, framework, args, python_api_us
                 elif 'example_inputs' in args:
                     raise AssertionError("'example_inputs' argument is not recognized, maybe you meant to provide 'example_input'?")
 
-                if 'use_legacy_frontend' in args and args['use_legacy_frontend']:
```
Comment (popovaan) [inline tools/mo/openvino/tools/mo/convert_impl.py:876]: We have tests for this functionality in https://github.com/openvinotoolkit/openvino/blob/eb178a753b5f0536b19efe597e805ef6634f16c6/tests/layer_tests/mo_python_api_tests/test_mo_convert_pytorch.py#L104

Please remove **use_legacy_frontend** flag in tests

Comment (popovaan) [inline tools/mo/openvino/tools/mo/convert_impl.py:876]: Please remove description of this functionality in documentation
https://github.com/openvinotoolkit/openvino/blob/eb178a753b5f0536b19efe597e805ef6634f16c6/docs/MO_DG/prepare_model/convert_model/Convert_Model_From_PyTorch.md?plain=1#L52

**Pull request merged**